BlockEntity/BEFruitpress.cs

```diff
@@ -390,28 +385,35 @@ private bool InteractMashContainer(IPlayer byPlayer, BlockSelection blockSel)
                     return false;
                 }
 
-
-                float transferableLitres;
+                float transferableLitres = (float)handStack.Attributes.GetDecimal("juiceableLitresLeft");
+                float usedLitres = (float)handStack.Attributes.GetDecimal("juiceableLitresTransfered");
                 int removeItems;
                 if (hprops.LitresPerItem == null)
                 {
-                    var availableLitres = (float)handStack.Attributes.GetDecimal("juiceableLitresLeft");
-                    transferableLitres = (float)Math.Min(availableLitres, juiceableLitresCapacity - juiceableLitresLeft);
-                    // the juiceableLitresLeft is per item if we have a stack of multiple
-                    // so we only remove one item and add the liters for one
-                    removeItems = 1;
+                    // the juiceableLitresLeft and juiceableLitresTransfered are per item if we have a stack of multiple
+                    // so we check if the press is full and only remove and add the liters for one item at a time
+                    if (juiceableLitresLeft + juiceableLitresTransfered + transferableLitres + usedLitres > juiceableLitresCapacity)
+                    {
+                        (Api as ICoreClientAPI)?.TriggerIngameError(this, "fullcontainer", Lang.Get("Container is full, press out juice and remove the mash before adding more"));
+                        return false;
+                    }
+
+                    if (juiceableLitresLeft + juiceableLitresTransfered <= 0) mashStack.Attributes.SetDouble("squeezeRel", handStack.Attributes.GetDouble("squeezeRel", 1));

```

Comment:
Sets the mash in the press to be as squeezed as the one you're adding by hand. This is specifically so that if you’re adding a wet mash to the empty press it will be as squeezed as it was before. If there’s already something in the press it will reset the squeeze entirely.